cirq-google/cirq_google/named_topologies.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import dataclasses
+import warnings
+from dataclasses import dataclass
+from functools import lru_cache
+from typing import Dict, List, Tuple, Any, Sequence
+
+import cirq
+import networkx as nx
+from matplotlib import pyplot as plt
+
+
+def cache(user_function):
```

Comment:
nit: missing type annotation

(Note that `Any` is better than nothing at all since missing type annotation disables type checking of the function's body. That said, here you can probably do better than `Any`, e.g. `Callable`?)

Reply:
We can automate this by passing [--disallow-untyped-defs](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-disallow-untyped-defs) to mypy.

Reply:
if and when we move to python 3.9+ you better hold on to your hats because this is exactly how the function is defined in functools sans type annotation :)

Reply:
@vtomole someone would have to fix all the untyped defs before enabling that flag

Reply:
Ah, I haven't checked how many there are. Hopefully it's not overwhelming.

Reply:
overwhelming is exactly the word I'd use to describe it

Reply:
Yeah, that word sums it up 😭 . Filed https://github.com/quantumlib/Cirq/issues/4393

Reply:
done